### PR TITLE
Changed: extract pandas DataFrame converter into a separate module

### LINKED ISSUE #215: extract pandas dataframe -> vizzu data obj conversion into separate package/module
It could be useful in other environments without the whole ipyvizzu package.
E.g. in py-script the user could have a dataframe to pass as an input for the JS Vizzu lib or a future PyScript-vizzu plugin.

### PR DESCRIPTION
**Pull request by veghdev**
- [x] separate pandas converter into an independent module
- [x] remove pandas from dependencies, add to extras
- [x] update the docs

resolve #215

### CODE REVIEW

File: src/ipyvizzu/data/converters/pandas_converter.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                return pd.DataFrame()
+            raise TypeError(
+                "df must be an instance of pandas.DataFrame or pandas.Series"
+            )
+        except ImportError as error:
+            raise ImportError(
+                "pandas is not available. Please install pandas to use this feature."
+            ) from error
+
+    def get_series_list_from_columns(self) -> List[Dict[str, Union[str, List[Any]]]]:
```
Comment (schaumb) [inline src/ipyvizzu/data/converters/pandas_converter.py:64]: I think this return hint is complicated enough to create type aliases

File: src/ipyvizzu/data/converters/pandas_converter.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            from pandas.api.types import (  # pylint: disable=import-outside-toplevel
+                is_numeric_dtype,
+            )
+
+            if is_numeric_dtype(column.dtype):
+                return self._get_measure_column_data(column)
+            return self._get_dimension_column_data(column)
+        except ImportError as error:
+            raise ImportError(
+                "pandas is not available. Please install pandas to use this feature."
```
[the remark this answers is not in the document]
Comment (veghdev) [reply src/ipyvizzu/data/converters/pandas_converter.py:115]: OK, the duplication is the problem.

File: src/ipyvizzu/data/converters/pandas_converter.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    def _get_df(self, df: Union["pd.DataFrame", "pd.Series"]) -> "pd.DataFrame":  # type: ignore
+        try:
+            import pandas as pd  # pylint: disable=import-outside-toplevel
+
+            if isinstance(df, pd.DataFrame):
+                return df
+            if isinstance(df, pd.Series):
+                return pd.DataFrame(df)
+            if df is None:
```
[the remark this answers is not in the document]
Comment (veghdev) [reply src/ipyvizzu/data/converters/pandas_converter.py:60]: @schaumb improved data type hints in [20760f](https://github.com/vizzuhq/ipyvizzu/pull/252/commits/20760f0c98d985bfd7ed26e510c1d67fa8a4aadd)

File: src/ipyvizzu/data/typing_alias.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+DimensionValue = str
+
+MeasureValue = Union[int, float]
+
+NestedMeasureValues = Union[MeasureValue, List["NestedMeasureValues"]]
+
+Value = Union[DimensionValue, MeasureValue]
+
+RecordValues = Union[DimensionValue, MeasureValue]
```
Comment (schaumb) [inline src/ipyvizzu/data/typing_alias.py:16]: why is it in plural?

File: src/ipyvizzu/data/converters/pandas_converter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (veghdev) [reply src/ipyvizzu/data/converters/pandas_converter.py]: fixed in [https://github.com/vizzuhq/ipyvizzu/pull/252/commits/7ba32a96fabf5532f45173160db264474667e613](7ba32a)

File: src/ipyvizzu/data/converters/pandas_converter.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (schaumb) [reply src/ipyvizzu/data/converters/pandas_converter.py]: I think It is clearer if the type is  `Optional[Union["pd.DataFrame", "pd.Series"]]`.

[the remark this answers is not in the document]
Comment (veghdev) [reply src/ipyvizzu/data/converters/pandas_converter.py]: OK, replaced

File: src/ipyvizzu/animation.py [hunk 5]
```diff
@@ -285,81 +286,90 @@ def add_data_frame(
...
+        default_measure_value: Optional[Any] = 0,
+        default_dimension_value: Optional[Any] = "",
     ) -> None:
         """
-        A method for adding data frame's index to an existing
+        [Deprecated] Add a `pandas` `DataFrame` or `Series` to an existing
         [Data][ipyvizzu.animation.Data] class instance.
 
+        This function is kept for backward compatibility
+        and calls the `add_df` method with the same arguments.
```
[the remark this answers is not in the document]
Comment (veghdev) [reply src/ipyvizzu/animation.py]: fixed in [389d7e](https://github.com/vizzuhq/ipyvizzu/pull/252/commits/389d7e3be362641c5efed3f980d4596a0139fa05)

**Review by schaumb — DISMISSED** [review]

**Review by schaumb — APPROVED** [review]

**Pull request merged**